app/scripts/modules/core/src/managed/overview/artifact/utils.ts

```diff
@@ -0,0 +1,127 @@
+import { DateTime } from 'luxon';
+
+import { ManagedWriter, showModal, useApplicationContext } from 'core/index';
+import { timeDiffToString } from 'core/utils';
+
+import { MarkAsBadActionModal, PinActionModal, UnpinActionModal } from './ActionModal';
+import { VersionAction } from './VersionMetadata';
+import { useFetchApplicationLazyQuery } from '../../graphql/graphql-sdk';
+import { QueryArtifactVersion, QueryLifecycleStep } from '../types';
+
+export const getLifecycleEventDuration = (
```

Comment:
Would it provide some safety to add unit tests to this helper functions?

Reply:
I'll add it to my todos but I don't think it's blocking